src/Lib/Metric.php

```diff
@@ -381,6 +442,44 @@ protected static function getVariableLabels(): array {
 		return $labels;
 	}
 
+	/**
+	 * Get rate metrics based on the base
+	 * @param  array<string,int>  $metrics
+	 * @return array<string,int>
+	 */
+	protected function getRateMetrics(array $metrics): array {
+		$now = time();
+		$duration = 0;
+		$saveSnapshot = ($now - $this->snapshotTime) > 0;
+		if ($saveSnapshot) {
+			$duration = $now - $this->snapshotTime;
+		}
+
+		$rateMetrics = [];
+		foreach ($metrics as $name => $value) {
+			$this->add($name, $value);
+
+			// Caculate rate metrics when we need
+			if (!str_starts_with($name, 'command_') || $duration <= 0) {
+				continue;
+			}
+
+			$rateName = "{$name}_rate";
```

Comment:
I would change `rate` to `rate_per_second` for better clarity.

Reply:
As disucssed changed to `_rps`